spirv_msl.cpp

```diff
@@ -428,6 +480,16 @@ void CompilerMSL::build_implicit_builtins()
 		set_extended_decoration(var_id, SPIRVCrossDecorationResourceIndexPrimary, msl_options.buffer_size_buffer_index);
 		buffer_size_buffer_id = var_id;
 	}
+
+	if (needs_view_mask_buffer())
+	{
+		uint32_t var_id = build_constant_uint_array_pointer();
+		set_name(var_id, "spvViewMask");
+		// This should never match anything.
+		set_decoration(var_id, DecorationDescriptorSet, kViewMaskBufferBinding);
+		set_decoration(var_id, DecorationBinding, msl_options.view_mask_buffer_index);
+		view_mask_buffer_id = var_id;
```

Comment:
This needs to have set_extended_decoration() with SPIRVCrossResourceIndexPrimary now as well. I had to fix the automatic resource binding assignment scheme, and it now assigns resources automatically since fallbacks to DecorationBinding are not correct in all cases.